app/views/income_additional/edit.html.erb

```diff
@@ -0,0 +1,33 @@
+<% content_for :header_title, 'Money & Income' %>
+
+<div class="form-card">
+  <header class="form-card__header">
+    <div class="step-section-header">
+      <div class="step-section-header__subhead">
+        Tell us more about your additional income.
+      </div>
+    </div>
+  </header>
+
+  <div class="form-card__content">
+    <%  income_fields = {
+      income_unemployment_insurance: "Unemployment Insurance",
+      income_ssi_or_disability: "SSI or Disability",
+      income_workers_compensation: "Worker's Compensation",
+      income_pension: "Pension",
+      income_social_security: "Social Security",
+      income_child_support: "Child Support",
+      income_foster_care: "Foster Care or Adoption Subsidies",
+      income_other: "Other Income"
+    }.stringify_keys.slice(*@additional_income) %>
+
+    <%= form_for @step, as: :step, builder: MbFormBuilder, url: current_path, method: :put do |f| %>
+
+      <% income_fields.each do |method, label| %>
+        <%= f.mb_input_field method, label, notes: ["Monthly amount"] %>
+      <% end %>
```

Comment:
Moved the ruby code to the controller here: 6322f2a

still iterating since the fields we are showing is dynamic (so hardcoding would require a lot of ruby conditionals in the view) but I think this is a clarifying improvement nonetheless!